### PR TITLE
Disable reliance on support of 1GB pages

### PR DESCRIPTION
**Pull request by christianchristensen**
Issue #685

Description of changes: (see issue)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by acatangiu** [comment]
Oh, also please sign your commit and provide a commit message that describes the change from code/functionality point of view.

You can change the existing commit's message/description and sign it at the same time by doing `commit -s --amend` on your local branch.

File: x86_64/src/regs.rs [hunk 1]
```diff
@@ -193,11 +195,21 @@ fn setup_page_tables(mem: &GuestMemory, sregs: &mut kvm_sregs) -> Result<()> {
     // Puts PML4 right after zero page but aligned to 4k.
     let boot_pml4_addr = GuestAddress(layout::PML4_START);
     let boot_pdpte_addr = GuestAddress(layout::PDPTE_START);
+    let boot_pde_addr = GuestAddress(layout::PDE_START);
 
     mem.write_obj_at_addr(boot_pdpte_addr.offset() as u64 | 0x03, boot_pml4_addr)
```
Comment (acatangiu) [inline x86_64/src/regs.rs]: Please add a comment for the 0-512 MB virtual address region.

```suggestion
    // Entry covering VA [0..512GB)
    mem.write_obj_at_addr(boot_pdpte_addr.offset() as u64 | 0x03, boot_pml4_addr)
```

File: x86_64/src/regs.rs [hunk 2]
```diff
@@ -311,6 +323,9 @@ mod tests {
 
         assert_eq!(0xa003, read_u64(&gm, layout::PML4_START));
         assert_eq!(0x83, read_u64(&gm, layout::PDPTE_START));
```
Comment (acatangiu) [inline x86_64/src/regs.rs]: This is the offending line that wasn't properly pulled from the crosvm fix.

```suggestion
        assert_eq!(0xb003, read_u64(&gm, layout::PDPTE_START));
```

Comment (mlctrez) [reply x86_64/src/regs.rs]: @acatangiu  my bad.. first time doing anything with rust

File: x86_64/src/regs.rs [hunk 3]
```diff
@@ -311,6 +323,9 @@ mod tests {
...
+        for i in 0..512 {
+            assert_eq!((i << 21) + 0x83u64, read_u64(&gm, (0xb000 + i * 8) as usize));
```
Comment (kamyuentse) [inline x86_64/src/regs.rs]: Use constant `PDE_START` instead of magic number `0xb000

File: x86_64/src/regs.rs (hunk 2 above)
Comment (kamyuentse) [inline x86_64/src/regs.rs]: Change `0x83` -> `0xb000 | 0x03`, see https://github.com/firecracker-microvm/firecracker/pull/731/files#diff-d2945d10fb0ec3da17729c614ac604b4R204

**Comment by christianchristensen** [comment]
@acatangiu Done.

**Pull request merged**